### DISCUSSION TITLE
TypeRegistrar and TypeResolver should be included

### DISCUSSION THREAD

**Discussion by Lasamat** [spectreconsole/spectre.console#915] [General]
**Is your feature request related to a problem? Please describe.**
I looked at the documentation and examples for using Dependancy Injection with this library and both the example for Logging and DI use the exact same two wrapper classes to convert from Microsoft.Extensions.DependencyInjection to use with this Library.

**Describe the solution you'd like**
I would like to have either those two classes already in the Library or a way to simply pass the ServiceCollection in the first place.

**Describe alternatives you've considered**
Alternative would be to copy paste those two exact same classes for all my DI Console Apps...

**Comment by nils-a** [comment]
@Lasamat They are not included to not "push" users of *Spectre.Console* into using one or the other DI framework. The `TypeRegistrar` and `TypeResolver` have been specifically implemented to be independent of specific DI frameworks.

The code you referred to is one possible implementation, [the *Cake* project, for example, uses another](https://github.com/cake-build/cake/blob/a1893a2bba26f49e089d66beeb78421cedf6126d/src/Cake/Infrastructure/Composition/AutofacTypeRegistrar.cs).
I, myself, do like *SimpleInjector* and don't use the Microsoft DI Extensions very often. In my projects, I use the [Spectre.Console.Registrars.SimpleInjector NuGet package](https://www.nuget.org/packages/Spectre.Console.Registrars.SimpleInjector) to not always copy-paste the same code.

**Reply by Lasamat** [reply]
@nils-a Thank you for the fast response. I'm relatively new to collab on Open Source and proper software architecture.

The SimpleInjector looks nice. If its okay I would like to PR it into the Spectre.Console.Cli -> CommandApp -> DI section of the documentation.

**Reply by nils-a** [reply]
Sure, good idea.

**Comment by hughesjs** [comment]
Might I suggest an alternative of including them in a `Spectre.Cli.DependencyInjection.Microsoft` package or something of that ilk?
